src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/Threads.java

```diff
@@ -151,6 +151,7 @@ private static synchronized void initialize(TypeDataBase db) {
 
         if (!VM.getVM().isCore()) {
             virtualConstructor.addMapping("CompilerThread", CompilerThread.class);
+            virtualConstructor.addMapping("TrainingReplayThread", TrainingReplayThread.class);
```

Comment:
The new SA TrainingReplayThread class is not needed since it only overrides isHiddenFromExternalView() to return true. You can instead use HiddenJavaThread.class here.

Reply:
Done